File: Solutions/p3.py
def isPrime(num):
    for i in range(2, int(num**.5)+1):
        if num%i==0:
            return False
    return True


dividend = 600851475143
for num in range((dividend//3) + 1, 3, -1):
    if dividend%num==0 and isPrime(num):
        break


print(num)

File: Solutions/p3.java
public class p3{
    public static boolean isPrime(Long num){
        for(int i=2; i<=(int) Math.pow(num, .5)+1; i++){
            if(num%i==0){
                return false;
            }
        }
        return true;
    }
    public static void main(String[] args){
        long dividend =  600851475143L;
        long primeFactor = -1;

        for(long i=3; i<=(dividend/3); ++i){
            if(dividend%i==0 && isPrime(dividend/i)){
                primeFactor = dividend/i;
                break;
            }
        }

        System.out.println(primeFactor);
    }
}